### PR TITLE
Release 2.0.11.4.0

### PR DESCRIPTION
**Pull request by kantai**
This PR would release 2.0.11.4.0 into master.

This software update is a point-release to change the transaction selection
logic in the default miner to prioritize by an estimated fee rate.  This release's
chainstate directory is compatible with chainstate directories from 2.0.11.3.0.

## Added

- FeeEstimator and CostEstimator interfaces. These can be controlled
  via node configuration options. See the `README.md` for more
  information on the configuration.
- New fee rate estimation endpoint `/v2/fees/transaction` (#2872). See
  `docs/rpc/openapi.yaml` for more information.

## Changed

- Prioritize transaction inclusion in blocks by estimated fee rates (#2859).

# Release Checklist

- [x] The release must be timed so that it does not interfere with a prepare phase.
- [x] Version number determined
- [x] No PRs are open that block this release
- [x] PR opened
- [x] `CHANGELOG.md` updated
- [x] Release candidate tagged
- [x] Tested on:
        Stacks Foundation staging environments.
        Hiro PBC testnet network.
        Hiro PBC mainnet mock miner.
- [x] RC announced in the #stacks-core-devs channel in the Stacks Discord.
- [x] RC successfully synced from genesis
- [x] No bugs or issues discovered
- [x] Tag additional stacks-blockchain team members to review the develop -> master PR.
- [ ] Once reviewed and approved, the release manager merges the PR, and tags the release

### CODE REVIEW

**Comment by jcnelson** [comment]
So I spun this up last week but was an idiot and had `miner = false` in the config file.  So I only have data from this afternoon.  But, it is encouraging, looking at the maximum-computed epoch fees by the miner versus those in the current node software:

```
37080: better (16730159 <= 24679446)
37081: better (11443634 <= 12915286)
37082: better (3295635 <= 6300000)
37083: worse (22254482 > 22184533)
37084: better (6958531 <= 9413376)
37085: better (2990618 <= 11568865)
37086: better (8676590 <= 9006808)
37087: better (15355205 <= 15732797)
37088: better (1359600 <= 16527128)
37089: better (11565525 <= 16671194)
37090: worse (12527976 > 10729745)
37091: better (471200 <= 471200)
37092: better (1605465 <= 9234814)
37093: better (4533740 <= 6800492)
37094: better (720162 <= 720162)
37095: better (1175122 <= 21025922)
37096: better (19434834 <= 31144482)
37097: worse (6791359 > 6246625)
37098: better (10075496 <= 23448609)
37099: better (18627012 <= 23320714)
37100: worse (17756396 > 17624490)
37101: better (17820948 <= 26196949)
37102: better (11110631 <= 18564401)
37103: better (7123890 <= 11830400)
37104: better (5939342 <= 12118320)
37105: better (7414541 <= 10595600)
37106: better (2720400 <= 6049500)
37107: better (11684961 <= 14589277)
37108: better (4300800 <= 10795117)
37109: better (8242639 <= 8762317)
37110: worse (3023987 > 2942787)
37111: worse (7804576 > 7684542)
37112: better (846030 <= 2248701)
```

**Comment by jcnelson** [comment]
Looks like we're good

```
37080: better (16730159 <= 24679446)
37081: better (11443634 <= 12915286)
37082: better (3295635 <= 6300000)
37083: worse (22254482 > 22184533)
37084: better (6958531 <= 9413376)
37085: better (2990618 <= 11568865)
37086: better (8676590 <= 9006808)
37087: better (15355205 <= 15732797)
37088: better (1359600 <= 16527128)
37089: better (11565525 <= 16671194)
37090: worse (12527976 > 10729745)
37091: better (471200 <= 471200)
37092: better (1605465 <= 9234814)
37093: better (4533740 <= 6800492)
37094: better (720162 <= 720162)
37095: better (1175122 <= 21025922)
37096: better (19434834 <= 31144482)
37097: worse (6791359 > 6246625)
37098: better (10075496 <= 23448609)
37099: better (18627012 <= 23320714)
37100: worse (17756396 > 17624490)
37101: better (17820948 <= 26196949)
37102: better (11110631 <= 18564401)
37103: better (7123890 <= 11830400)
37104: better (5939342 <= 12118320)
37105: better (7414541 <= 10595600)
37106: better (2720400 <= 6049500)
37107: better (11684961 <= 14589277)
37108: better (4300800 <= 10795117)
37109: better (8242639 <= 8762317)
37110: worse (3023987 > 2942787)
37111: worse (7804576 > 7684542)
37112: better (846030 <= 2248701)
37113: better (2312125 <= 7814750)
37114: better (8411349 <= 11860355)
37115: better (3280923 <= 4685006)
37116: better (417600 <= 696000)
37117: worse (1155360 > 720000)
37118: better (1737200 <= 3505510)
37119: better (2846867 <= 3177587)
37120: better (4475976 <= 6109865)
37121: better (1556469 <= 7320602)
37122: worse (9878330 > 8525055)
37123: better (12092632 <= 16960286)
37124: better (8619410 <= 9844338)
37125: better (12732587 <= 13717712)
37126: better (12797390 <= 13094990)
37127: better (2000000 <= 3474671)
37128: better (4657589 <= 7848797)
37129: better (4766632 <= 5657032)
37130: better (5580287 <= 14099087)
37131: better (7711120 <= 19043025)
37132: better (11117851 <= 11426999)
37133: better (1752580 <= 2274760)
37134: better (2986028 <= 3270428)
37135: worse (2906867 > 2731584)
37136: better (2691464 <= 10245125)
37137: better (10526815 <= 16751052)
37138: better (6407286 <= 21397426)
37139: better (4612464 <= 24367519)
37140: better (10479279 <= 20363855)
37141: better (5371591 <= 11226980)
37142: better (6470755 <= 6917955)
37143: better (3961225 <= 5319225)
37144: better (4269708 <= 5548573)
37145: better (2061827 <= 3212627)
37146: worse (3176939 > 2607265)
37147: worse (6479805 > 5071805)
37148: better (15857359 <= 21884274)
37149: better (5433158 <= 8722995)
37150: worse (4258662 > 3943265)
37151: better (12436701 <= 16283285)
37152: better (7124641 <= 8321041)
37153: better (1690400 <= 4321600)
37154: better (681200 <= 9777964)
37155: better (7070865 <= 9754635)
37156: better (6713234 <= 13446342)
37157: better (7806423 <= 9805633)
37158: better (1079377 <= 2754302)
37159: better (3883156 <= 5289733)
37160: better (4470400 <= 8477067)
37161: better (12796732 <= 15201975)
37162: better (6552337 <= 9221464)
37163: better (2097276 <= 4762327)
37164: better (1929600 <= 3086912)
37165: better (11078491 <= 13698215)
37166: better (850000 <= 2244732)
37167: better (11405014 <= 12150014)
37168: better (2326580 <= 3815280)
37169: better (2325256 <= 3844960)
37170: worse (13058366 > 5021630)
```

**Review by jcnelson — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,28 @@ All notable changes to this project will be documented in this file.
...
+rate instead of raw fee. This release's chainstate directory is
+compatible with chainstate directories from 2.0.11.3.0.
+
+## Added
+
+- FeeEstimator and CostEstimator interfaces. These can be controlled
+  via node configuration options. See the `README.md` for more
+  information on the configuration.
+- New fee rate estimation endpoint `/v2/fees/transaction` (#2872). See
+  `docs/rpc/openapi.yaml` for more information.
```
Comment (jcnelson) [inline CHANGELOG.md:21]: Forgot to add that #2869 was applied here (mmap'ed MARF connections)

**Comment by jcnelson** [comment]
Your PR comment doesn't say that #2869 was merged in, but the changelog does (nbd, just fyi)

**Comment by kantai** [comment]
> Your PR comment doesn't say that #2869 was merged in, but the changelog does (nbd, just fyi)

Ah, yes -- the rationale was that the top matter of the CHANGELOG entry should just summarize the major behavior changes. MMAP at this point is just a performance improvement, so I figured it'd be okay for that to just be mentioned in the meat of the changelog.

**Review by gregorycoppola — APPROVED** [review]
Thanks!

**Pull request merged**